pkg/obj/pipeline.go

```diff
@@ -74,15 +75,15 @@ func NewPipeline(wrd *WorkflowRunDeps) *Pipeline {
 
 func ConfigurePipeline(ctx context.Context, p *Pipeline) error {
 	if err := p.Deps.WorkflowRun.Own(ctx, p); err != nil {
-		return err
+		return fmt.Errorf("failed to own Pipeline: %w", err)
```

Comment:
FYI, I think this might break some of the checks for Sentry downwind of this. Take a look at `reconciler/filter/errorcapture.go` for example. There's no good enforcement or test for this right now but it might be better to preemptively wrap everything instead:

```go
return errmark.MapLast(err, func(err error) error {
  return fmt.Errorf("failed to own Pipeline: %w", err)
}
```

Reply:
I wasn't going to comment, but I'd prefer we don't contextualize these errors... at least not at this time.

Reply:
I added the wrapping because it was very hard to get a clear picture of what was going wrong. Almost every error message in the logs was "Err: object required", so I had to start wrapping everything to figure out what pieces in `pkg/obj` were failing.

I can roll these back, but they seem super valuable to me. Would unwrapping them for the sentry checks work? https://godoc.org/github.com/pkg/errors#Unwrap

Reply:
I realize that, but in general we should be relying upon the original errors as much as possible. We may wrap these in cleaner errors later, but this should take the form of more organized errors if we do. Adding a bit of extra context via formatting is not particularly clean. And this is - as you've noted - primarily helpful for people when dealing with manually debugging the service, not production-level code. It just speaks of adding in debug level statements.

Reply:
For more context, there are several specific reasons for not doing this, including the following:
1. These messages are essentially noting what is happening in that particular code, not representing anything about the actual error. An error/exception should specifically note what the error is in as much explicit detail as possible. These generic messages are moving in the wrong direction, even if they temporarily help gain in understanding while debugging.
2. Many temporary/transient errors are being requeued as part of the controller. These are not necessarily "failures" and any language specifically noting these as failures may make it more difficult to debug in many situations (particularly in production if someone is dealing with an oncall issue and is not familiar with this code).
3. In cases like these (and exactly this case), this is usually a sign that underlying problems need addressed, and putting in short-term debugging-level information is not addressing the root cause. The "Err: object required" is mostly seen while the controller is still progressing through applying objects. This particular error message is a side effect of the load behavior and itself may need specific improvement.
4. In most cases, except perhaps on the highest level (and even then done carefully), errors should be passed on as-is without wrapping or contextualizing these further. The locality and purity of the original error should be respected.